framework/caching/Cache.php

```diff
@@ -468,6 +505,7 @@ protected function setValues($data, $duration)
             if ($this->setValue($key, $value, $duration) === false) {
                 $failedKeys[] = $key;
             }
+            $this->endProfile($key, __METHOD__);
```

Comment:
Yes, current cache class tests are located in https://github.com/yiisoft/yii2/blob/master/tests/framework/caching/ these should be extended so that they run once with `enableProfiling = true` and once with `enableProfiling = false` to test these variations.
Here is how to run tests: https://github.com/yiisoft/yii2/tree/master/tests#yii-20-unit-tests

Let us know if you need help on this.